crates/matrix-sdk-base/src/rooms/normal.rs

```diff
@@ -575,39 +575,55 @@ impl Room {
                 }
             }
 
-            (names, None)
+            (names, None, None)
         } else {
             let mut joined_members = self.members(RoomMemberships::JOIN).await?;
 
             // Make the ordering deterministic.
             joined_members.sort_unstable_by(|lhs, rhs| lhs.name().cmp(rhs.name()));
 
-            // We can make a good prediction of the total number of members here. This might
-            // be incorrect if the database info is outdated.
+            // We can make a good prediction of the total number of joined members here.
+            // This might be incorrect if the database info is outdated.
             let num_joined = Some(joined_members.len());
 
-            (
-                joined_members
-                    .into_iter()
-                    .take(NUM_HEROES)
-                    .filter_map(|u| (u.user_id() != own_user_id).then(|| u.name().to_owned()))
-                    .collect(),
-                num_joined,
-            )
+            let mut invited_members = self.members(RoomMemberships::INVITE).await?;
+
+            // Make the ordering deterministic.
+            invited_members.sort_unstable_by(|lhs, rhs| lhs.name().cmp(rhs.name()));
+
+            // We can make a good prediction of the total number of invited members here.
+            // This might be incorrect if the database info is outdated.
+            let num_invited = Some(invited_members.len());
+
+            let heroes = joined_members
+                .into_iter()
+                .chain(invited_members)
+                .filter(|u| u.user_id() != own_user_id)
+                .take(NUM_HEROES)
```

Comment:
It's a bit wasteful to do extra SQL queries to fetch invited members, at the risk of disposing them just a bit later here (because of the `take()`). Can we be smarter and either 1. use a single `self.members` as requested above, or 2. do the second `members()` call iff joined_members.len()+1 < NUM_HEROES?